### DISCUSSION TITLE
OrmLite - SQL BulkInsert

### DISCUSSION THREAD

**Discussion by dylanvdmerwe** [ServiceStack/Discuss#30] [Ideas]
Are there any plans to inckude some kind of BulkInsert operation for Ormlite?

Use Case
This method is very useful if you are inserting multiple rows towards the database in a very speedy manner. It is high-performant in nature as it is using the real bulk operation natively from [ADO.NET](http://ado.net/) (via SqlBulkCopy class).

If you are working to insert range of rows from 1000 or more, then use this method.

WIP: https://gist.github.com/zelid/6965002

**Comment by mythz** [comment]
Thanks for submitting, we'll update this thread when there's progress on this feature request.

**Comment by mythz** [comment]
[Support for RDBMS Bulk Inserts](https://docs.servicestack.net/releases/v6_10#rdbms-bulk-inserts) have been added for all supported RDBMS behind OrmLite's new `BulkInsert` API in the latest ServiceStack v6.10 Release:

```csharp
db.BulkInsert(rows);
```

Please see the [Release Notes](https://docs.servicestack.net/releases/v6_10#rdbms-bulk-inserts) for more info.

Which also includes a number of benchmarks we ran measuring the performance vs Single Inserts in the [Which RDBMS has the fastest Bulk Insert implementation?](https://servicestack.net/posts/bulk-insert-performance) Blog Post.
